module13and14/LoopCanvas/LoopCanvas/Views/HomeView.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  HomeView.swift
+//  LoopCanvas
+//
+//  Created by Peter Rice on 6/25/24.
+//
+
+import SwiftUI
+
+struct HomeView: View {
+  @StateObject var canvasStore = CanvasStore()
+  var body: some View {
+    NavigationStack {
+      List {
+        NavigationLink {
+          let canvasViewModel = CanvasViewModel(
+            canvasModel: CanvasModel(),
+            musicEngine: AudioKitMusicEngine(),
+            canvasStore: canvasStore)
+          CanvasView(viewModel: canvasViewModel)
```

Comment:
@waltertyree so it turns out that this CanvasViewI() + CanvasViewModel() are evaluated when rendering the list, even before the NavigationLink is actually tapped. This is problematic b/c I do a bunch of initialization in CanvasModel, CanvasViewModel(), and AudioKitMusicEngine() which ideally would only be done once when the view is loaded. This is an even bigger issue for the SongListView where - b/c I create a NavigationLink for each saved song in your documents directory - an in that case - a CanvasViewModel() for each of your saved songs is created - before you choose which one you want to load. Obviously this is not ideal.

It seems that this pattern of creating a ViewModel object and injecting it into a details view via the initialized in the NavigationLink closure is somewhat incompatible with the way NavigationLinks eagerly evaluate their destination views. What's the idiomatic way to handle this w/ views that depend on view models which use NavigationLinks ?

Reply:
Ok - turns out that going back to the .navigationDestination() view modifier solves the problem b/c its contents are evaluated lazily when the user taps on a link. To use this on home screen I just need to add an enum value for the links to avoid typo's w raw strings.